src/Medidata.ZipkinTracer.Core.Collector/SpanProcessor.cs

```diff
@@ -53,71 +59,73 @@ public virtual void Stop()
-            }
-            catch (TException)
+            if ((serializableSpans.Count() >= maxBatchSize)
+                || (serializableSpans.Any() && spanProcessorTaskFactory.IsTaskCancelled())
+                || (subsequentPollCount > MAX_NUMBER_OF_POLLS))
             {
-                if ( retries < 3 )
+                SerializableSpan serializableSpan;
+                var listOfSpans = new List<SerializableSpan>();
+                while (serializableSpans.TryDequeue(out serializableSpan))
+                {
+                    listOfSpans.Add(serializableSpan);
+                }
+                try
                 {
-                    retries++;
-                    Log(client, logEntries);
+                    if(listOfSpans.Any()) 
+                        SendSpansToZipkin(JsonConvert.SerializeObject(listOfSpans));
                 }
-                else
+                catch (WebException ex)
                 {
-                    throw;
+                    logger.Error("Failed to send to zipking with error: " + ex.Message);
                 }
             }
         }
 
-        private LogEntry Create(Span span)
+        public virtual void SendSpansToZipkin(string requestBody)
         {
-            var spanAsString = Convert.ToBase64String(ConvertSpanToBytes(span));
-            return new LogEntry()
+            using (var client = new WebClient())
             {
-                Category = "zipkin",
-                Message = spanAsString
-            };
-        }
+                var tracerPath = "/api/v1/spans";
+                UriBuilder uriBuilder = new UriBuilder();
+                uriBuilder.Host = zipkinServer;
+                uriBuilder.Port = zipkinPort;
+                uriBuilder.Scheme = zipkinPort == 443 ? "https" : "http";
 
-        private byte[] ConvertSpanToBytes(Span span)
-        {
-            var buf = new MemoryStream();
-            TProtocol protocol = protocolFactory.GetProtocol(new TStreamTransport(buf, buf));
-            span.Write(protocol);
-            return buf.ToArray();
+                try
+                {
+                    client.BaseAddress = uriBuilder.Uri.ToString();
+                    client.UploadString(tracerPath, "POST", requestBody);
+                    logger.Info("Sending Spans to Zipkin Success");
+                }
+                catch (WebException ex)
+                {
+                    var response = ex.Response as HttpWebResponse;
+                    if ((response != null))
+                    {
+                        // read all response info
+                        var responseStream = response.GetResponseStream();
+                        if (responseStream != null)
+                        {
+                            var responseString = new StreamReader(responseStream).ReadToEnd();
+                            logger.Error("Sending Spans to Failed with error: " + ex.Message + " and message:" + responseString);
```

Comment:
Maybe hard to understand the exception log - the error is the message and the message is the response string. How about `Failed to send spans to Zipkin server (HTTP status code returned: " + response.StatusCode + "). Exception message: " + ex.Message + ", response from server: " + responseString`